Assets/MixedRealityToolkit.SDK/Inspectors/UX/ManipulationHandler/ManipulationHandlerInspector.cs

```diff
@@ -4,23 +4,261 @@
+                }
+            }
+
+            EditorGUILayout.Space();
+            twoHandedFoldout = EditorGUILayout.Foldout(twoHandedFoldout, "Two Handed Manipulation", true);
+
+            if (twoHandedFoldout)
+            {
+                if (handedness == ManipulationHandler.HandMovementType.TwoHandedOnly ||
+                    handedness == ManipulationHandler.HandMovementType.OneAndTwoHanded)
+                {
+                    EditorGUILayout.PropertyField(twoHandedManipulationType);
+                }
+                else
+                {
+                    EditorGUILayout.HelpBox("Two handed manipulation disabled. If you wish to enable two handed manipulation select it as a Manipulation Type above.", MessageType.Info);
+                }
+            }
+
+            var mh = (ManipulationHandler)target;
+            var rb = mh.GetComponent<Rigidbody>();
+
+            EditorGUILayout.Space();
+            physicsFoldout = EditorGUILayout.Foldout(physicsFoldout, "Physics", true);
+
+            if (physicsFoldout)
+            {
+                if (rb != null)
+                {
+                    EditorGUILayout.PropertyField(releaseBehavior);
+                }
+                else
+                {
+                    EditorGUILayout.HelpBox("Physics options disabled. If you wish to enable physics options, add a Rigidbody component to this object.", MessageType.Info);
+                }
+            }
+
+            EditorGUILayout.Space();
+            constraintsFoldout = EditorGUILayout.Foldout(constraintsFoldout, "Constraints", true);
+
+            if (constraintsFoldout)
+            {
+                EditorGUILayout.PropertyField(constraintOnRotation);
+                EditorGUILayout.PropertyField(constraintOnMovement);
+            }
+
+            EditorGUILayout.Space();
+            smoothingFoldout = EditorGUILayout.Foldout(smoothingFoldout, "Smoothing", true);
+
+            if (smoothingFoldout)
+            {
+                EditorGUILayout.PropertyField(smoothingActive);
+                EditorGUILayout.PropertyField(smoothingAmountOneHandManip);
+            }
+
+            EditorGUILayout.Space();
+            eventsFoldout = EditorGUILayout.Foldout(eventsFoldout, "Manipulation Events", true);
+
+            if (eventsFoldout)
+            {
+                EditorGUILayout.PropertyField(onManipulationStarted);
+                EditorGUILayout.PropertyField(onManipulationEnded);
+                EditorGUILayout.PropertyField(onHoverEntered);
+                EditorGUILayout.PropertyField(onHoverExited);
+            }
+
+            // reset foldouts style
+            style.fontStyle = previousStyle;
+
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void Migrate()
+        {
+            var mh1 = target as ManipulationHandler;
```

Comment:
I think it would be better to move the Migrate function into a MigrationUtility or other public space, so we can use it for a batch conversion as well. 
Not part of this PR: might be good to add a Migration Utility Window for MRTK where you can trigger upgrading to the new UX elements in current scene / all scenes / prefabs etc.